test/bugs/1719.js

```diff
@@ -0,0 +1,52 @@
+// https://github.com/swagger-api/swagger-js/issues/1719
+
+import path from 'path';
+import fs from 'fs';
+import jsYaml from 'js-yaml';
+
+import resolveSubtree from '../../src/subtree-resolver';
+
+const testDoc = jsYaml.safeLoad(
+  fs.readFileSync(path.join('test', 'data', 'swos-123-ref-object-reference.yaml'), 'utf8')
+);
+
+test('1719: $ref object should not be treated as a reference (string)', async () => {
+  const specPath = ['paths', '/throwError', 'post'];
+  const res = await resolveSubtree(testDoc, specPath);
+
+  expect(res.errors.length).toEqual(0);
+
+  expect(res.spec).toEqual(
```

Comment:
This is very hard to read and will be very hard to maintain. Please see example of [subtree-resolver test](https://github.com/swagger-api/swagger-js/blob/3b5cc18fd7c738f3954609b0820459ef57e15c11/test/subtree-resolver.js#L558), and how we use deep non-strict equality to test more complicated structures.

Reply:
The intent was to not fully describe the expected resolved subtree, and only check for a specific deeply nested value. However, it may still be simplified by not making it as deeply nested. I'll take a look at refactoring.

Reply:
In that case this describes your intention more properly:
```js
expect(houseForSale).toHaveProperty(['prop1', 'innerProp1', ...., '$ref', 'type'], 'string');

Reply:
`toHaveProperty` implemented.